cmd/sdn-cni-plugin/openshift-sdn_linux.go

```diff
@@ -234,11 +234,30 @@ func (p *cniPlugin) CmdAdd(args *skel.CmdArgs) error {
 			}
 		}
 
+		// add routes from config.clusterNetworks array here with overlay mtu.
+		routeMTU := int(config.MTU) - 50
+
+		link, err = netlink.LinkByName(args.IfName)
+		if err != nil {
+			return fmt.Errorf("failed to get link: %v via SDN: %v", args.IfName, err)
+		}
+		for _, cn := range config.ClusterNetworks {
+			route := &netlink.Route{
+				Dst:       cn.ClusterCIDR,
+				LinkIndex: link.Attrs().Index,
+				MTU:       routeMTU,
+			}
+			if err := netlink.RouteAdd(route); err != nil && !os.IsExist(err) {
+				return fmt.Errorf("failed to add mtu based route to dst: %v via SDN: %v", cn.ClusterCIDR, err)
+			}
+		}
+
 		dsts = append(dsts, serviceIPNet)
 		for _, dst := range dsts {
 			route := &netlink.Route{
 				Dst: dst,
 				Gw:  defaultGW,
+				MTU: routeMTU,
```

Comment:
Some service traffic will end up going to nodes, not pods... will that screw anything up if we get a spurious ICMP in response to that? (I guess probably not since the ICMP would be in response to traffic that had been SNATted to a node, so it presumably wouldn't make it back to the pod anyway...)

Reply:
We should be ok.   This traffic flow is identical to the problem we're trying to solve.  What's different this time is that if the 'bug' occurs, when `ip_forward()` is done after DNAT from node to pod address, the pod `eth0` mtu (and that of `tun0`) will now be that of the underlay (e.g. 1500), so an ICMP will not be generated.